### ISSUE TITLE
[BUG] GDFError: GDF_VALIDITY_MISSING in binary ops where one column has nulls, other doesn't

### ISSUE THREAD

**Issue by kkraus14** [rapidsai/cudf#696]
Minimal reproducer:
```python
import cudf
import numpy as np
import pandas as pd
arr_1 = np.array([np.nan, np.nan, 1., 2. ,3.])
arr_2 = np.array([2., 2., 2., 2. ,2.])

# pandas
pdf =  pd.DataFrame()
pdf['arr_1'] = arr_1
pdf['arr_2'] = arr_2
pd_out = pdf['arr_1'] / pdf['arr_2']

# cudf
df = cudf.dataframe.DataFrame()
df['arr_1'] = arr_1
df['arr_2'] = arr_2

# This throws the error
cudf_out = df['arr_1'] / df['arr_2']
```

**Comment by kkraus14** [comment]
Note: this is because we don't allocate a validity mask if there's no nulls in a column, so libcudf needs to handle the case where the validity mask is nullptr as all valid.

**Comment by devavret** [comment]
Coincidentally, I just found this bug on my own an hour ago but when I went to report it, found it here.

**Comment by kkraus14** [comment]
Fixed by #1019
